source/common/router/rds_impl.cc

```diff
@@ -174,5 +183,18 @@ Router::RouteConfigProviderSharedPtr RouteConfigProviderManagerImpl::getRouteCon
   return new_provider;
 };
 
+Http::Code RouteConfigProviderManagerImpl::handlerRoutes(const std::string&,
+                                                         Buffer::Instance& response) {
+  for (auto provider : routeConfigProviders()) {
+    response.add(fmt::format(
+        "route_config_name: {}\n",
+        std::static_pointer_cast<RdsRouteConfigProviderImpl>(provider)->route_config_name_));
```

Comment:
Seems a `configAsString()` message on the `routeConfigProviderr` woudl avoid this casting.